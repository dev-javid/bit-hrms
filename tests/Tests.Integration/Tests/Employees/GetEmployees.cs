namespace Tests.Integration.Tests.Employees
{
    public class GetEmployees(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/employees";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Employees/GetEmployees.sql");
            using HttpResponseMessage response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
