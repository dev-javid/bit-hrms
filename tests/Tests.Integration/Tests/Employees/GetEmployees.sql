INSERT INTO public.employees
(id, company_id, user_id, department_id, job_title_id, first_name, last_name, date_of_birth, date_of_joining, father_name, phone_number, company_email, personal_email, address, city, pan, aadhar, created_by, created_on, modified_by, modified_on)
VALUES
(3, 999, 300, 3000, 4000, 'Test-First-3', 'Test-Last-3', '2000-05-03', '2024-05-03', 'Test-Father-3', '01234567899', 'test3@example.com', 'test6@example.com', 'Test-Address', 'Test-City', 'XXXXX0000X', '000000000000', 0, '2024-05-03 03:31:32.227954-07', NULL, NULL),
(4, 999, 200, 3000, 4000, 'Test-First-4', 'Test-Last-4', '2000-05-03', '2024-05-03', 'Test-Father-4', '01234567899', 'test4@example.com', 'test7@example.com', 'Test-Address', 'Test-City', 'XXXXX1111X', '111111111111', 0, '2024-05-03 03:31:32.227954-07', NULL, NULL);