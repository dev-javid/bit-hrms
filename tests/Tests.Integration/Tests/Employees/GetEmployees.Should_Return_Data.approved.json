{
    "page": 1,
    "limit": 10,
    "total": 4,
    "items": [
        {
            "employeeId": 1,
            "departmentId": 3000,
            "departmentName": "Test-Department",
            "jobTitleId": 4000,
            "jobTitleName": "Test-Department",
            "firstName": "Test-First-1",
            "lastName": "Test-Last-1",
            "fullName": "Test-First-1 Test-Last-1",
            "dateOfBirth": "2000-05-03",
            "dateOfJoining": "2024-05-03",
            "fatherName": "Test-Father-1",
            "phoneNumber": "01234567890",
            "companyEmail": "test1@example.com",
            "personalEmail": "test@example.com",
            "address": "Test-Address",
            "city": "Test-City",
            "pan": "XXXXX0000X",
            "aadhar": "000000000000",
            "documents": []
        },
        {
            "employeeId": 2,
            "departmentId": 3000,
            "departmentName": "Test-Department",
            "jobTitleId": 4000,
            "jobTitleName": "Test-Department",
            "firstName": "Test-First-2",
            "lastName": "Test-Last-2",
            "fullName": "Test-First-2 Test-Last-2",
            "dateOfBirth": "2000-05-03",
            "dateOfJoining": "2024-05-03",
            "fatherName": "Test-Father-2",
            "phoneNumber": "01234567890",
            "companyEmail": "test2@example.com",
            "personalEmail": "test@example.com",
            "address": "Test-Address",
            "city": "Test-City",
            "pan": "XXXXX1111X",
            "aadhar": "111111111111",
            "documents": []
        },
        {
            "employeeId": 3,
            "departmentId": 3000,
            "departmentName": "Test-Department",
            "jobTitleId": 4000,
            "jobTitleName": "Test-Department",
            "firstName": "Test-First-3",
            "lastName": "Test-Last-3",
            "fullName": "Test-First-3 Test-Last-3",
            "dateOfBirth": "2000-05-03",
            "dateOfJoining": "2024-05-03",
            "fatherName": "Test-Father-3",
            "phoneNumber": "01234567899",
            "companyEmail": "test3@example.com",
            "personalEmail": "test6@example.com",
            "address": "Test-Address",
            "city": "Test-City",
            "pan": "XXXXX0000X",
            "aadhar": "000000000000",
            "documents": []
        },
        {
            "employeeId": 4,
            "departmentId": 3000,
            "departmentName": "Test-Department",
            "jobTitleId": 4000,
            "jobTitleName": "Test-Department",
            "firstName": "Test-First-4",
            "lastName": "Test-Last-4",
            "fullName": "Test-First-4 Test-Last-4",
            "dateOfBirth": "2000-05-03",
            "dateOfJoining": "2024-05-03",
            "fatherName": "Test-Father-4",
            "phoneNumber": "01234567899",
            "companyEmail": "test4@example.com",
            "personalEmail": "test7@example.com",
            "address": "Test-Address",
            "city": "Test-City",
            "pan": "XXXXX1111X",
            "aadhar": "111111111111",
            "documents": []
        }
    ]
}